lib/active_support/cache/memcached_store.rb

```diff
@@ -130,11 +129,11 @@ def cas(name, options = nil)
 
         handle_exceptions(return_value_on_error: false) do
           instrument(:cas, name, options) do
-            @connection.cas(key, expiration(options)) do |raw_value|
-              entry = deserialize_entry(raw_value)
-              value = yield entry.value
+            @connection.cas(key, expiration(options), !options[:raw]) do |raw_value|
+              value = options[:raw] ? raw_value : deserialize_entry(raw_value).value
```

Comment:
The `raw` option should only be used for serialization.  On deserialization, we should rely on the memcached flags to determine if deserialization is needed.  That way we can avoid accidentally deserializing a raw string.

Reply:
Right sorry, I think I was trying to get a failing test to pass. I'll have a look again.

Reply:
Hmm wait, so should we support `raw` on `cas` then or not? We can only pass one `decode` value to `Memcached#cas`, which it uses for both `multi_get` _and_ `multi_cas`. If we pass `true`, it will call the coder on both encodes and decodes.

So it seems like here, we can't avoid hitting the coder on raw writes. Or am I reading this wrong? 🤔